api/src/main/java/org/openmrs/module/pihcore/htmlformentry/action/RegisterBabyAction.java

```diff
@@ -0,0 +1,192 @@
+            Concept yesConcept = Context.getConceptService().getConceptByMapping(YES_CONCEPT, "CIEL");
+            Concept noConcept = Context.getConceptService().getConceptByMapping(NO_CONCEPT, "CIEL");
+
+            Patient patient = formEntrySession.getPatient();
+            Encounter encounter = formEntrySession.getEncounter();
+            boolean updateEncounter = false;
+            for (Obs candidate : encounter.getObsAtTopLevel(false)) {
+                if (candidate.getConcept().equals(newbornDetailsConcept)) {
+                    Set<Obs> groupMembers = candidate.getGroupMembers();
+                    if ( groupMembers != null ) {
+                        for (Obs groupMember : groupMembers) {
+                            // search for Labour and Delivery obs that have the "Register patient in EMR" obs set to No (SL-617)
+                            if (groupMember.getConcept().equals(registerBabyConcept) && groupMember.getValueCoded().equals(noConcept)) {
+                                //find the gender and the birthdate time of the baby
+                                String gender = getObsValue(groupMembers, BABY_GENDER_CONCEPT_SOURCE, BABY_GENDER_CONCEPT);
+                                Date birthDatetime = getObsDateValue(groupMembers, DATE_TIME_OF_BIRTH_CONCEPT_SOURCE, DATE_TIME_OF_BIRTH_CONCEPT);
+                                if (StringUtils.isNotBlank(gender) && (birthDatetime != null)) {
+                                    Patient baby = registerBaby(patient, gender, birthDatetime, encounter.getLocation());
+                                    if (baby != null && baby.getUuid() != null) {
+                                        groupMember.setValueCoded(yesConcept);
+                                        groupMember.setComment(baby.getUuid());
+                                        updateEncounter = true;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            if (updateEncounter) {
+                Context.getEncounterService().saveEncounter(encounter);
+            }
+
+            //automatically redirect to the children page
+            formEntrySession.setAfterSaveUrlTemplate("pihcore/children/children.page?patientId={{patient.id}}");
+        }
+    }
+
+    Patient registerBaby(Patient mother, String gender, Date birthDatetime, Location location) {
+        PersonService personService = Context.getPersonService();
+        RelationshipType motherToChildRelationshipType = personService.getRelationshipTypeByUuid(PihEmrConfigConstants.RELATIONSHIPTYPE_MOTHERTOCHILD_UUID);
+        RelationshipType relationship = personService.getRelationshipTypeByUuid(motherToChildRelationshipType.getUuid());
+        List<Relationship> relationships = new ArrayList<Relationship>();
```

Comment:
Are there circumstances we need to handle here where the obs says that the baby hasn't been registered yet, but in fact they have?  Specifically for retrospective entry?  Do we want to sanity check that the mother doesn't have an existing relationship to a baby with the same birthdate before creating a new patient here?

Reply:
I thought we discussed those problems last week, and we decided to redirect the user to the children page where they will be able to remove any duplicates in the system, or search for another patient record?